packages/DataTable/src/hooks/useBestTableDimensions.ts

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import debounce from "lodash.debounce";
+import convertSizeStringToNumber, { Direction } from "../helpers/convertSizeStringToNumber";
+
+type Dimensions = {
+  width: number;
+  height: number;
+  shouldHaveHorizontalScroll: boolean;
+};
+
+export default function useBestTableDimensions({
+  tableRef,
+  componentRef,
+  height: maxHeight,
+  width: maxWidth,
+  shouldResizeWithViewport,
+}: {
+  tableRef: React.RefObject<HTMLDivElement>;
+  componentRef: React.RefObject<unknown>;
+  height: string;
+  width: string;
+  shouldResizeWithViewport: boolean;
+}): Dimensions {
+  const [dimensions, setDimensions] = React.useState<Dimensions>({
+    width: convertSizeStringToNumber(maxWidth, Direction.width),
+    height: convertSizeStringToNumber(maxHeight, Direction.height),
+    shouldHaveHorizontalScroll: false,
+  });
+
+  const resetDimension = React.useCallback(() => {
+    if (!tableRef.current) return;
+
+    const theadEl = tableRef.current.querySelector('[data-pka-anchor="dataTable.thead"]');
+    const tbodyEl = tableRef.current.querySelector('[data-pka-anchor="dataTable.tbody"]');
+
+    if (!theadEl || !tbodyEl) {
+      return;
+    }
+
+    setDimensions(() => {
+      const realWidth = theadEl.clientWidth;
+      const realHeight = theadEl.clientHeight + tbodyEl.clientHeight;
+      const maxWidthInNumber = convertSizeStringToNumber(maxWidth, Direction.width);
+      const maxHeightInNumber = convertSizeStringToNumber(maxHeight, Direction.height);
+
+      return {
+        width: Math.min(maxWidthInNumber, realWidth),
+        height: Math.min(maxHeightInNumber, realHeight),
+        shouldHaveHorizontalScroll: maxWidthInNumber < realWidth,
+      };
+    });
+  }, [maxHeight, maxWidth, tableRef]);
+
+  React.useLayoutEffect(() => {
+    resetDimension();
+    const debouncedReset = debounce(resetDimension, 300);
```

Comment:
Had to use lodash function since the `useCallbackDebouncer` is for a different usage, see https://github.com/acl-services/paprika/blob/80361b22f7c8c81f716131966979a6673f0d49ea/packages/ActionBar/src/components/SearchInput/SearchInput.js#L19

Reply:
Maybe the `300` can be extracted to a `const` at the top?